test/unit/RelayServer.test.ts

```diff
@@ -390,68 +472,134 @@ describe('RelayServer tests', function () {
 
         expect(maxPossibleGas.toString()).to.be.eq(
           (
-            fakeEstimationBeforeFees +
-            fakeEstimationBeforeFees * FAKE_GAS_FEE_PERCENTAGE
+            FAKE_ESTIMATION_BEFORE_FEES +
+            FAKE_ESTIMATION_BEFORE_FEES * FAKE_GAS_FEE_PERCENTAGE
           ).toString()
         );
       });
 
-      it('Should charge fees based on transfer value, when transferFeePercentage > 0 and a transfer() is being relayed', async function () {
-        const fakeServerConfigParams = {
-          app: {
-            gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
-            transferFeePercentage: FAKE_TRANSFER_FEE_PERCENTAGE,
-          },
-        };
-        sinon.stub(relayServer, 'config').value(fakeServerConfigParams);
+      describe('When transferFeePercentage > 0', function () {
+        it('Should charge fees based on transfer value, when  and a transfer() is being relayed', async function () {
+          const fakeServerConfigParams = {
+            app: {
+              gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
+              transferFeePercentage: FAKE_TRANSFER_FEE_PERCENTAGE,
+            },
+          };
+          sinon.stub(relayServer, 'config').value(fakeServerConfigParams);
 
-        const maxPossibleGas = await relayServer.getMaxPossibleGas({
-          relayRequest: {
-            request: {
-              tokenContract: constants.AddressZero,
-              tokenAmount: TOKEN_AMOUNT_IN_REQUEST,
-              data: dataWhenTransfer,
+          const maxPossibleGas = await relayServer.getMaxPossibleGas({
+            relayRequest: {
+              request: {
+                tokenContract: constants.AddressZero,
+                tokenAmount: TOKEN_AMOUNT_IN_REQUEST,
+                data: dataWhenTransfer,
+              },
+              relayData: {
+                gasPrice: GAS_PRICE,
+              },
             },
-            relayData: {
-              gasPrice: GAS_PRICE,
+          } as EnvelopingTxRequest);
+
+          expect(maxPossibleGas.toString()).to.be.eq(
+            BigNumberJs(FAKE_ESTIMATION_BEFORE_FEES)
+              .plus(expectedFeeFromTransfer)
+              .toString()
+          );
+        });
+
+        it('Should not charge fees if the amount to transfer is zero when a transfer() is being relayed', async function () {
+          const dataWhenTransferingZero = dataWhenTransfer.replace(
+            tokenAmountToTransferAsHex,
+            '0'.repeat(64)
+          );
+
+          const fakeServerConfigParams = {
+            app: {
+              gasFeePercentage: FAKE_GAS_FEE_PERCENTAGE,
+              transferFeePercentage: FAKE_TRANSFER_FEE_PERCENTAGE,
             },
-          },
-        } as EnvelopingTxRequest);
+          };
+          sinon.stub(relayServer, 'config').value(fakeServerConfigParams);
```

Comment:
We could move this in the `beforeEach` section

Reply:
Done.